### ISSUE TITLE
Limit type of windows that get animated?

### ISSUE THREAD

**Issue by nkkollaw** [BlackCapCoder/compton#4]
All windows seem to get animated, including tooltips and other small elements that aren't properly windows.

Is there a way to specify the type of windows that get animated, being able to exclude tooltips, panels, etc.?

**Comment by BlackCapCoder** [comment]
No, but it totally should be. Compton actually has a option for this called `wintypes` where you can override fade, shadow and opacity based on window type.

**Comment by BlackCapCoder** [comment]
Yes, I obviously want to integrate with existing compton features whenever possible.

**Comment by nkkollaw** [comment]
...I was also thinking about the problem, and was wondering if instead of adding to Compton, it would be ppssible to create a script that works on top of Compton and just takes care of the animation..?

**Comment by BlackCapCoder** [comment]
Not without scripting support in compton I'm afraid. Compton is using the X-server API to register as a composite manager and to do its rendering, and as far as I know X only has a C API.
If slightly worse performance is acceptable to you you might like another composite manager called Ortle. Its codebase is really clean and well written, so hacking it is a breeze.

**Comment by BlackCapCoder** [comment]
I pushed a version I made with window transitions in case you'd like to try it: https://github.com/BlackCapCoder/ortle

**Comment by nkkollaw** [comment]
Ha! Yes, very cool.

It does has some problems, like my Xfce panel disappears when I launch it, but it's really cool.

I get right now there is no solution :-/

**Comment by nkkollaw** [comment]
...your fork of ortle seems very promising, though. Is there any way that you know of to add shadows to that instead of fixing Compton?

**Comment by BlackCapCoder** [comment]
Let's continue this discussion on the ortle fork: https://github.com/BlackCapCoder/ortle/issues/1

**Comment by alexhulbert** [comment]
Has this issue been dropped in favor of the ortle fork? I'd still love to see this feature in your compton fork. With spawn-center disabled and transition-length turned down, compton is stable enough for me to use this feature on my main laptop. However, in order for the transitions to not be annoying on things like tooltips, I'd have to turn it off for a few window classes and ```window_type```s.

**Comment by nkkollaw** [comment]
@alexhulbert I don't think so, @BlackCapCoder was just generously trying to help me out by finding an alternative solution.

**Comment by BlackCapCoder** [comment]
@alexhulbert No, I'll get to it eventually, but anything in compon happens at a snails pace unfortunately.

**Comment by nkkollaw** [comment]
@alexhulbert : how do you do "spawn-center disabled and transition-length turned down"..?

**Comment by nkkollaw** [comment]
...people seem to want this feature: https://github.com/chjj/compton/issues/231/. There is also a reference to this fork :-)
